packages/flutter_tools/lib/src/device.dart

```diff
@@ -67,46 +67,11 @@ class PlatformType {
 }
 
 /// A class to get all available devices.
-class DeviceManager {
+abstract class DeviceManager {
 
   /// Constructing DeviceManagers is cheap; they only do expensive work if some
   /// of their methods are called.
-  List<DeviceDiscovery> get deviceDiscoverers => _deviceDiscoverers;
-  final List<DeviceDiscovery> _deviceDiscoverers = List<DeviceDiscovery>.unmodifiable(<DeviceDiscovery>[
```

Comment:
This was getting hit in devices_test and failing if it grabbed the wrong context, I did some minor cleanup but it needs to be fixed later so that there is no accidental caching